aten/src/ATen/native/cuda/Shape.cu

```diff
@@ -226,6 +226,39 @@ __global__ void CatArrayBatchedCopy_contig(
     }
 }
 
+
+template <typename T, typename IndexType, int Dims, int batch_size, int stride_size, int alignment, int elems_per_vec>
+__global__ void CatArrayBatchedCopy_vectorized(
+    char* output,
+    CatArrInputTensorMetadata<T, IndexType, batch_size, stride_size> inputs,
+    TensorSizeStride<IndexType, CAT_ARRAY_MAX_INPUT_DIMS> os,
+    const int concatDim,
+    IndexType trailingSize) {
+
+    IndexType tid = blockIdx.x * blockDim.x + threadIdx.x;
```

Comment:
maybe needs a cast to `IndexType`? (our old friend)

Reply:
we come here only if all can be done in int32 but I can do it, just to be safe

Reply:
on second thought, I'll leave it like this for now, because it's in all the kernels, so we should change that in a separate PR